development/process.md

```diff
@@ -101,6 +102,93 @@ keep the amount of rework from the contributor to a minimum.
+necessarily a single commit. Unrelated changes should generally be extracted
+into their own PRs.
+
+If a pull request contains a stack of more than one commit, then popping any
+number of commits from the top of the stack, should not break the PR, meaning
+that  every commit should build and pass all tests.
+
+Commit messages and history are important as well, because they are used by
+other developers to keep track of the motivation behind changes. Keep logical
+diffs grouped together in separate commits and order commits in a way that
+explains by itself the evolution of the change. Rewriting and reordering commits
+is a natural part of the review process.
+
+Mechanical changes like refactoring, renaming, removing duplication, extracting
+helper methods, static imports should be kept separated from logical and
+functional changes like adding a new feature or modifying code behaviour. This
+makes reviewing the code much easier and reduces the chance of introducing
+unintended changes in behavior.
+
+Whenever in doubt on splitting a change into a separate commit, ask yourself the
+following question: if all other work in the PR needs to be reverted after
+merging to master for some objective reason, such as a bug has been discovered,
+is it worth keeping that commit still in master.
+
+When writing a commit message, follow the seven rules of a great commit message:
+
+* Separate subject from body with a blank line.
+* Limit the subject line to 50 characters.
+* Capitalize the subject line
+* Do not end the subject line with a period.
+* Use the imperative mood in the subject line.
+* Wrap the body at 72 characters.
+* Use the body to explain what and why versus how.
+
+Read the [full commit message guide](https://chris.beams.io/posts/git-commit/)
+for more details and examples.
+
+## Release note guidelines <a name="release-note"></a>
+
+Release notes should communicate the user-facing aspects of all change with a
+specific version. As a results internal refactorings or improvements, such as
+build fixes or new tests, are not included. New features, new configuration,
+performance changes, bug fixes and other aspects relevant for users and
+administrators must be included.
+
+Release note entries should be suggested by the contributor. Correct content and
+wording is part of the pull request review and merge process. If necessary the
+contributor and merging maintainer can work with the maintainer assembling the
+release notes PR.
+
+Some release notes are broken up into separate section for different components.
+For example, Trino release are ordered following the
+[template](https://github.com/trinodb/trino/blob/master/docs/release-template.md)
+
+In each section different release notes entries are sorted:
+
+1. **New features**: Start with `Add` or `Add support for`
+2. **Performance improvements**: Start with `Improve` or `Improve performance`
+3. **Bug fixes**: Start with `Fix` or `Prevent`
```

Comment:
Are prefixes hard rules or suggestions. We are not following those now, and I think just these set may be a bit limiting, to make everything sound naturally.

Reply:
Those are all guidelines and suggestions